packages/next-intl/src/middleware/middleware.tsx

```diff
@@ -89,8 +88,10 @@ export default function createMiddleware<Locales extends AllLocales>(
               configWithDefaults.localePrefix === 'as-needed' &&
               urlObj.pathname.startsWith(`/${locale}`)
             ) {
-              const regex = new RegExp(`\\/${locale}\\b`);
```

Comment:
This was already handled in `getNormalizedPathname` but I noticed we missed this code path here.